solidity/contracts/test/TestTBTCDepositorProxy.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.0;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+
+import "../integrator/TBTCDepositorProxy.sol";
+import "../integrator/IBridge.sol";
+import "../integrator/ITBTCVault.sol";
+
+contract TestTBTCDepositorProxy is TBTCDepositorProxy {
+    event InitializeDepositReturned(uint256 depositKey);
+
+    event FinalizeDepositReturned(uint256 tbtcAmount, bytes32 extraData);
+
+    function initialize(address _bridge, address _tbtcVault) external {
+        __TBTCDepositorProxy_initialize(_bridge, _tbtcVault);
+    }
+
+    function initializeDepositPublic(
+        IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+        IBridgeTypes.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) external {
+        uint256 depositKey = _initializeDeposit(fundingTx, reveal, extraData);
+        emit InitializeDepositReturned(depositKey);
+    }
+
+    function finalizeDepositPublic(uint256 depositKey) external {
+        (uint256 tbtcAmount, bytes32 extraData) = _finalizeDeposit(depositKey);
+        emit FinalizeDepositReturned(tbtcAmount, extraData);
+    }
+
+    function calculateTbtcAmountPublic(
+        uint64 depositAmountSat,
+        uint64 depositTreasuryFeeSat
+    ) external view returns (uint256) {
+        return _calculateTbtcAmount(depositAmountSat, depositTreasuryFeeSat);
+    }
+}
+
+contract MockBridge is IBridge {
+    using BTCUtils for bytes;
+
+    mapping(uint256 => IBridgeTypes.DepositRequest) internal _deposits;
+
+    uint64 internal _depositTxMaxFee = 1 * 1e7; // 0.1 BTC
+
+    event DepositRevealed(uint256 depositKey);
+
+    function revealDepositWithExtraData(
+        IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+        IBridgeTypes.DepositRevealInfo calldata reveal,
+        bytes32 extraData
+    ) external {
+        bytes32 fundingTxHash = abi
+            .encodePacked(
+                fundingTx.version,
+                fundingTx.inputVector,
+                fundingTx.outputVector,
+                fundingTx.locktime
+            )
+            .hash256View();
+
+        uint256 depositKey = uint256(
+            keccak256(
+                abi.encodePacked(fundingTxHash, reveal.fundingOutputIndex)
+            )
+        );
+
+        require(
+            _deposits[depositKey].revealedAt == 0,
+            "Deposit already revealed"
+        );
+
+        IBridgeTypes.DepositRequest memory request;
+
+        request.depositor = msg.sender;
+        request.amount = uint64(10 * 1e8); // 10 BTC
+        /* solhint-disable-next-line not-rely-on-time */
+        request.revealedAt = uint32(block.timestamp);
+        request.vault = reveal.vault;
+        request.treasuryFee = uint64(1 * 1e8); // 1 BTC
```

Comment:
Can we calculate it based on the funding deposit amount:

```suggestion
        request.treasuryFee = _depositTreasuryFeeDivisor > 0
            ? fundingOutputAmount / _depositTreasuryFeeDivisor
            : 0;
```

And make the divisor configurable:
```solidity
    function setDepositTreasuryFeeDivisor(uint64 value) external {
        _depositTreasuryFeeDivisor = value;
    }
```

Reply:
I proposed a change in https://github.com/keep-network/tbtc-v2/pull/779.